### PR TITLE
Improve dns-search validation, empty domains now return an error

### LINKED ISSUE #3426: Empty dns search option should set "search" option as empty in containers /etc/resolv.conf
/kind bug

**Description**

According to docs:
> (Use --dns-search=. if you don't wish to set the search domain)

As I understand this, by specifying ``--dns-search=''`` my search domain should be empty, no matter what it was before. It's not the real case.

**Steps to reproduce the issue:**

``podman run -it --rm --name f --dns-search='' alpine cat /etc/resolv.conf``
Ouput:
```
search redhat.com sshnaidm.remote
nameserver 192.168.2.1
nameserver 8.8.8.8
```
**Describe the results you received:**
The results are exactly if I didn't set dns-search in options at all:
```
podman run -it --rm --name f alpine cat /etc/resolv.conf
search redhat.com sshnaidm.remote
nameserver 192.168.2.1
nameserver 8.8.8.8
```
If I explicitly set to non-empty string, it works fine:
```
podman run -it --rm --name f --dns-search='search.com' alpine cat /etc/resolv.conf
search search.com
nameserver 192.168.2.1
nameserver 8.8.8.8
```

**Describe the results you expected:**
I'd expect:
```
nameserver 192.168.2.1
nameserver 8.8.8.8
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.4.0
RemoteAPI Version:  1
Go Version:         go1.11.10
OS/Arch:            linux/amd64

```

**Output of `podman info --debug`:**

```
  compiler: gc
  git commit: ""
  go version: go1.11.10
  podman version: 1.4.0
host:
  BuildahVersion: 1.8.3
  Conmon:
    package: podman-1.4.0-2.fc29.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 0.2.0, commit: 80f19b1591b03dbbc53bf28b154230c7ea372281'
  Distribution:
    distribution: fedora
    version: "29"
  MemFree: 1339047936
  MemTotal: 25142370304
  OCIRuntime:
    package: runc-1.0.0-93.dev.gitb9b6cc6.fc29.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc8+dev
      commit: 82f4855a8421018c9f4d74fbcf2da7f8ad1e11fa
      spec: 1.0.1-dev
  SwapFree: 12117864448
  SwapTotal: 12645822464
  arch: amd64
  cpus: 4
  hostname: sshnaidm.remote
  kernel: 5.1.11-200.fc29.x86_64
  os: linux
  rootless: true
  uptime: 33h 7m 40.8s (Approximately 1.38 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ConfigFile: /home/sshnaidm/.config/containers/storage.conf
  ContainerStore:
    number: 2
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mount_program=/usr/bin/fuse-overlayfs
  GraphRoot: /home/sshnaidm/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 32
  RunRoot: /tmp/1000
  VolumePath: /home/sshnaidm/.local/share/containers/storage/volumes

```
btw,
Inspection returns same results in case of DNS search is empty: ``"DNSSearch": [],``
Seems like it should be ``"DNSSearch": [''],``

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by jwhonce**
Fixes #3426

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podman/shared/create.go [hunk 1]
```diff
@@ -627,6 +627,11 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 		return nil, errors.Errorf("cannot pass additional search domains when also specifying '.'")
 	}
 
+	// Check for explicit dns-search domain of ''
+	if c.Changed("dns-search") && len(c.StringSlice("dns-search")) == 0 {
+		return nil, fmt.Errorf("'' is not a valid domain")
```
Comment (mheon) [inline cmd/podman/shared/create.go]: Nit: errors.Errorf more consistent with the rest of the package

**Comment by mheon** [comment]
One trivial nit, otherwise LGTM

**Comment by edsantiago** [comment]
LGTM but out of curiosity, does `--dns-search=.` (dot) work for anyone? It's documented in `podman-run(1)` but this is what I get, root or rootless:
```
$ podman run --rm --dns-search=. alpine_labels cat /etc/resolv.conf
Error: . is not a valid domain
```

**Comment by rhatdan** [comment]
```
          Set custom DNS search domains (Use --dns-search=. if you don't wish
       to set the search domain)
```
@edsantiago I have no idea what this means.  Might be a bug in the man page?

**Comment by rhatdan** [comment]
```
# docker run fedora cat /etc/resolv.conf
# Generated by NetworkManager
search bos.redhat.com redhat.com
nameserver 10.11.5.19
nameserver 10.5.30.160
nameserver 10.192.20.245
# NOTE: the libc resolver may not support more than 3 nameservers.
# The nameservers listed below may not be recognized.
nameserver 10.192.206.245
nameserver 10.200.0.246
sh-5.0# docker run --dns-search=. fedora cat /etc/resolv.conf
nameserver 10.11.5.19
nameserver 10.5.30.160
nameserver 10.192.20.245
nameserver 10.192.206.245
nameserver 10.200.0.246
```

**Comment by edsantiago** [comment]
OK so maybe that was copied blindly from the docker man page but without actually coding it? 

My reason for testing this is that #3426, the issue this PR proposes to address, is actually a little bit of a confusion. The report quotes the man page (with dot), but then proceeds to report an error with '' (empty string). I have no idea what's going on here.

**Comment by rhatdan** [comment]
@edsantiago man page was correct, code was mostly correct except for a little overzealous in our domain checking. Opened a PR to fix.

**Pull request merged**